lib/win32/certstore/store_base.rb

```diff
@@ -58,6 +58,26 @@ def cert_add(store_handler, cert_file_path)
         end
       end
 
+      def cert_delete(store_handler, certificate_name)
+        cert_name = FFI::MemoryPointer.new(2, 128)
+        begin
+          while (pCertContext = CertEnumCertificatesInStore(store_handler, pCertContext) and not pCertContext.null? ) do
+            if (CertGetNameStringW(pCertContext, CERT_NAME_FRIENDLY_DISPLAY_TYPE, CERT_NAME_ISSUER_FLAG, nil,cert_name, 1024) &&
+              cert_name.read_wstring.downcase == certificate_name.downcase)
+              if CertDeleteCertificateFromStore(CertDuplicateCertificateContext(pCertContext))
```

Comment:
Instead of using `CertEnumCertificatesInStore` which lists all the certificates and looping over it, we should use some function to find the particular certificate. e.g. [CertFindCertificateInStore](https://msdn.microsoft.com/en-us/library/windows/desktop/aa376064(v=vs.85).aspx) function gives the pointer to the `cert_context` structure, which can be passed to `CertDeleteCertificateFromStore`.
This is a slightly big change.